### ISSUE TITLE
Memory leak when copying large file (>2.5 GB)

### ISSUE THREAD

**Issue by lauxjpn** [billziss-gh/sshfs-win#252]
When copying large files (testes with files larger than 2.5 GB) from an SSHFS-Win mount to a local Windows NTFS drive, the process that copies the file (_not_ the `sshfs.exe` process and `not` the `ssh.exe` process) will allocate a lot of memory while copying the file (about the size of the file), but will not release it afterwards.

Therefore, after copying multiple large files, all available memory has been exhausted and system speed is down to a crawl.

In the following screenshot, a simple PowerShell `Copy-Item` command (PID 7500) was executed to copy a 2.5 GB file from an SSHFS-Win mount to a local Windows drive, resulting in over 4 GB or memory allocated, that would not get freed until the `powershell.exe` process (PID 7500) would be closed:

![image](https://user-images.githubusercontent.com/14178357/99460039-6931b880-292f-11eb-9d55-c7dc1826ebbf.png)

---

Version: `sshfs-win-3.5.20024-x64`

**Comment by billziss-gh** [comment]
From your description this appears to be a problem with Powershell rather than a problem with SSHFS. What happens if you try to perform a similar file copy on NTFS?

**Comment by lauxjpn** [comment]
> From your description this appears to be a problem with Powershell rather than a problem with SSHFS. What happens if you try to perform a similar file copy on NTFS?

When copying a file from an NTSF drive to an NTSF drive, the memory consumption of the `powershell.exe` process stays at about 50 MB.

I also tested this with `robocopy` (which is the initial scenario that brought our server down to its knees), where the same issue appears.

I can also test a simple Win32 `CopyFile` call, but I expect it to have the same result.
I also tried adding/removing the `-olarge_read -omax_readahead=1GB` options, but they have no effect on the issue.

**Comment by lauxjpn** [comment]
> From your description this appears to be a problem with Powershell rather than a problem with SSHFS.

@billziss-gh You are right.

I created a PowerShell job from within PowerShell, that manages the sshfs process (so that I can kill it after I am done, since it can get stuck when the connection closes unexpectedly and I would need to restart it then anyway).

The sshfs.exe command I used, contained the `-odebug -ologlevel=debug1` switches. The result was, that the out PowerShell process (that I use to run my script) was continuously increasing its memory usage, because it was reading the streams of the inner PowerShell process (that hosts the sshfs.exe process).

By removing the two options, the memory consumption is now fine and stable for the PowerShell process below 100 MB, below 15 MB for sshfs.exe and below 15 MB for robocopy.exe.
